File: 0338-counting-bits/0338-counting-bits.py
class Solution(object):
    def countBits(self, n):
        """
        :type n: int
        :rtype: List[int]
        """

        ans = [0]
        for i in range(1, n+1):
            decVal = i
            count = 0
            while decVal > 0:
                count += decVal & 1
                decVal >>= 1
            ans.append(count)
        
        return ans
        

File: 0338-counting-bits/0338-counting-bits.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        //SC: O(n)
        vector<int> ans;
        //TC: O(n*32)
        for(int j = 0; j <= n; j++){
            int count = 0;
            for(int i = 0; i < 32; i++){
                if(j & (1 << i)){
                    count++;
                }
            }
            ans.push_back(count);
        }
        return ans;
    }
};